pkg/composer/up_service.go

```diff
@@ -165,10 +165,15 @@ func (c *Composer) upServiceContainer(ctx context.Context, service *serviceparse
 		cmd.Stdin = os.Stdin
 	}
 
+	// Containers are always detached, even for `nerdctl compose up --detach=false`.
+	// (In that case, containers are detached and then the log viewer is "attached")
+	//
+	// See pkg/composer/serviceparser/serviceparser.go
 	if !container.Detached {
-		cmd.Stdout = os.Stdout
-		cmd.Stderr = os.Stderr
+		panic("container.Detached can't be false for compose mode")
```

Comment:
Should it return an error or panic directly?

Reply:
Intentionally panicking, because unlike other errors this error must not happen with any user operation.